### ISSUE TITLE
Cylc install does not copy file called log in bin

### ISSUE THREAD

**Issue by jfrost-mo** [cylc/cylc-flow#5223]
**Describe the bug**
When trying to run a program called `log` I get a "no such program" error. Upon investigation I found that the file had *not* been copied from `cylc-src/workflow/bin` to `cylc-run/workflow/run1/bin` when I ran `cylc install workflow`. Files with other names were copied just fine.

**Release version(s) and/or repository branch(es) affected?**
Cylc 8.0.3 installed at Met Office.

**Steps to reproduce the bug**
Make a test workflow that runs a program called "log".
Minimal flow.cylc:
```cylc
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    final cycle point = 1
    [[graph]]
        R1 = """
            test_log
        """
[runtime]
    [[test_log]]
        script = "log"
```
bin/log: (Make executable)
```bash
#! /bin/bash
echo "Test"
```

**Expected behaviour**
The `log` file to be copied, and for the workflow to run the program.

**Additional context**
Is this an intentional design feature to avoid copying logs? If so it should probably be documented somewhere.

**Comment by hjoliver** [comment]
Thanks @jfrost-mo 

Reproduced.

This is documented: https://cylc.github.io/cylc-doc/stable/html/user-guide/installing-workflows.html#admonition-1 

The reason for this is, installed workflows have an important top level directory called `log`, so a `log` in the source directory would clash with that.

However, the implementation should distinguish between `log` at the top level and in sub-directories! ... so yes this is a bug.

**Comment by wxtim** [comment]
Thank you @jfrost-mo  for reporting, and reporting in a way that makes understanding and reproducing the problem straightforward.

**Comment by dpmatthews** [comment]
Closed by #5224
